python/ray/rllib/env/vector_env.py

```diff
@@ -101,3 +113,81 @@ def vector_step(self, actions):
     @override(VectorEnv)
     def get_unwrapped(self):
         return self.envs
+
+
+class _RemoteVectorizedGymEnv(_VectorizedGymEnv):
+    """Internal wrapper for gym envs to implement VectorEnv as remote workers.
+    """
+
+    def __init__(self, make_env, num_envs,
+                 action_space=None, observation_space=None):
+        _VectorizedGymEnv.__init__(self, make_env, [], num_envs,
+                                   action_space, observation_space)
+
+        for env in range(self.envs):
+            assert isinstance(env, ray.actor.ActorClass), \
+                "Your environment needs to be ray remote environment"
```

Comment:
Instead of requiring the env to be decorated as remote, you could add the decorator here on the fly (i.e., ray.remote(cls)). That way, it becomes possible to toggle between remote / non-remote with just the use_remote flag.

Reply:
I was mentioning that problem in the description. We don't have the environment class, just a function that maps env_context to env object. Even if we wanted to create env object and deduce the class from environment object (which is very ugly), we would need to know how to map env_context to env constructor arguments, which is info that we do not have.

The only solution I can think of is changing register env interface as mentioned in pull request desription.

Reply:
Got it, this seems fine then.